arch/arm/soc/nordic_nrf5/nrf51/Kconfig.series

```diff
@@ -18,5 +18,6 @@ config  SOC_SERIES_NRF51X
 	select HAS_CMSIS
 	select HAS_NORDIC_MDK
 	select HAS_NORDIC_HAL
+	select HAS_FLASH_PAGE_LAYOUT
```

Comment:
rather than adding this to all the SoCs, can't we just add it to driver Kconfig?